### PR TITLE
Added implementation for okhttp

### PR DESCRIPTION
**Pull request by sirio7g**

### CODE REVIEW

File: stream-core/src/main/java/io/getstream/client/model/activities/BaseActivity.java [hunk 1]
```diff
@@ -50,27 +52,33 @@
...
     public void setId(String id) {
         this.id = id;
     }
 
+    @JsonIgnore
     public String getOrigin() {
         return origin;
     }
 
+    @JsonProperty("origin")
```
Comment (sirio7g) [inline stream-core/src/main/java/io/getstream/client/model/activities/BaseActivity.java]: JsonProperty is optional, why did you add them?

File: stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java [hunk 2]
```diff
@@ -177,6 +177,18 @@ public void deleteActivityById(BaseFeed feed, String activityId) throws IOExcept
...
     }
 
+    public void deleteActivityByForeignId(BaseFeed feed, String activityId) throws IOException, StreamClientException {
+        HttpDelete request = new HttpDelete(UriBuilder.fromEndpoint(baseEndpoint)
+                .path("feed").path(feed.getFeedSlug()).path(feed.getUserId()).path(activityId + "/")
+                .queryParam(StreamRepositoryImpl.API_KEY, apiKey)
+                .queryParam("foreign_id", Boolean.toString(true))
+                .build());
+        LOG.debug("Invoking url: '{}", request.getURI());
+        try (CloseableHttpResponse response = httpClient.execute(addAuthentication(feed, request), HttpClientContext.create())) {
```
Comment (sirio7g) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java]: I think you can use fireAndForget method

File: stream-repo-apache/src/main/java/io/getstream/client/repo/utils/SignatureUtils.java [hunk 3]
```diff
@@ -63,18 +63,17 @@ private SignatureUtils() {
      * @param activity  Activity to sign.
      */
     public static void addSignatureToRecipients(final String secretKey, final BaseActivity activity) throws StreamClientException {
-        if (activity.getTo() == null || activity.getTo().isEmpty()) {
```
Comment (sirio7g) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/utils/SignatureUtils.java]: Why did you remove this check?

Comment (tbarbugli) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/utils/SignatureUtils.java]: to field is optional

File: stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java (hunk 2 above)
Comment (tbarbugli) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java]: doesn't make it impossible to retry failed requests then?

File: stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java [hunk 4]
```diff
@@ -33,9 +35,9 @@
 
     @BeforeClass
     public static void setLog() {
-        System.setProperty("org.apache.commons.logging.Log", "org.apache.commons.logging.impl.SimpleLog");
-        System.setProperty("org.apache.commons.logging.simplelog.showdatetime", "true");
-        System.setProperty("org.apache.commons.logging.simplelog.log.org.apache.http", "DEBUG");
+//        System.setProperty("org.apache.commons.logging.Log", "org.apache.commons.logging.impl.SimpleLog");
```
Comment (sirio7g) [inline stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java]: Instead of commenting them out, you can just remove the annotation @BeforeClass

File: stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java [hunk 5]
```diff
@@ -130,16 +187,214 @@ public void shouldAddActivity() throws IOException, StreamClientException {
...
+        activity.setActor("actor");
+        activity.setObject("object");
+        activity.setTarget("target");
+        activity.setVerb("verb");
+        List<SimpleActivity> firstRequest = flatActivityService.getActivities().getResults();
+        assertThat(firstRequest.size(), is(0));
+        flatActivityService.addActivity(activity);
+        List<SimpleActivity> secondRequest = flatActivityService.getActivities().getResults();
+        assertThat(secondRequest.size(), is(1));
+        Assert.assertTrue(secondRequest.get(0).getId().matches("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"));
```
Comment (sirio7g) [inline stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java]: you can rewrite in:
```
assertThat(......, is(true));
```

File: stream-core/src/main/java/io/getstream/client/model/activities/BaseActivity.java (hunk 1 above)
Comment (tbarbugli) [inline stream-core/src/main/java/io/getstream/client/model/activities/BaseActivity.java]: that's the only way I could find to get jackson to do "asymmetrical" de/serialisation (origin field should never be sent but should be deserialised from responses) same applies for id and duration props

File: stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java (hunk 5 above)
Comment (tbarbugli) [inline stream-repo-apache/src/test/java/io/getstream/client/IntegrationTest.java]: good point :+1:

File: stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java (hunk 2 above)
Comment (sirio7g) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java]: I was referring to a method call: https://github.com/GetStream/stream-java/blob/master/stream-repo-apache/src/main/java/io/getstream/client/repo/StreamRepositoryImpl.java#L199
It will reduce duplicate code in this class.
Btw, on okhttp implementation retries is handled automatically by the HTTP client implementation: I set it to 3.

File: stream-core/src/main/java/io/getstream/client/model/feeds/Feed.java [hunk 6]
```diff
@@ -47,6 +47,9 @@
  */
 public interface Feed {
 
+    public String getId();
```
Comment (sirio7g) [inline stream-core/src/main/java/io/getstream/client/model/feeds/Feed.java]: Please add javadocs here

File: stream-core/src/main/java/io/getstream/client/model/feeds/Feed.java [hunk 7]
```diff
@@ -47,6 +47,9 @@
...
+    public String getToken();
```
Comment (sirio7g) [inline stream-core/src/main/java/io/getstream/client/model/feeds/Feed.java]: Here as well

File: stream-repo-apache/src/main/java/io/getstream/client/repo/utils/SignatureUtils.java (hunk 3 above)
Comment (sirio7g) [inline stream-repo-apache/src/main/java/io/getstream/client/repo/utils/SignatureUtils.java]: It's not optional in this specific context, isn't it?
